code/game/objects/structures/inflatable.dm

```diff
@@ -0,0 +1,292 @@
+	health -= damage
+	if(health <= 0)
+		user.visible_message("<span class='danger'>[user] tears open [src]!</span>")
+		deflate(TRUE)
+	else	//for nicer text~
+		user.visible_message("<span class='danger'>[user] tears at [src]!</span>")
+
+/obj/structure/inflatable/attack_alien(mob/user as mob)
+	if(islarva(user))
+		return
+	attack_inflatable(user, 15)
+
+/obj/structure/inflatable/attack_animal(mob/user as mob)
+	if(!isanimal(user))
+		return
+	var/mob/living/simple_animal/M = user
+	if(M.melee_damage_upper <= 0)
+		return
+	attack_inflatable(M, M.melee_damage_upper)
+
+/obj/structure/inflatable/attack_slime(mob/user as mob)
+	var/mob/living/simple_animal/slime/S = user
+	if(!S.is_adult)
+		return
+	attack_inflatable(user, rand(10, 15))
+
+/obj/structure/inflatable/attackby(obj/item/W as obj, mob/user as mob, params)
+	if(!istype(W))
+		return
+	if(is_pointed(W))
+		visible_message("<span class='danger'>[user] pierces [src] with [W]!</span>")
+		deflate(TRUE)
+	if(W.damtype == BRUTE || W.damtype == BURN)
+		hit(W.force)
+		..()
+	return
+
+/obj/structure/inflatable/proc/hit(var/damage, var/sound_effect = TRUE)
+	health = max(0, health - damage)
+	if(sound_effect)
+		playsound(loc, 'sound/effects/Glasshit.ogg', 75, 1)
+	if(health <= 0)
+		deflate(TRUE)
+
+/obj/structure/inflatable/AltClick()
+	if(usr.stat || usr.restrained())
+		return
+	if(!Adjacent(usr))
+		return
+	deflate()
+
+/obj/structure/inflatable/proc/deflate(var/violent = TRUE)
+	playsound(loc, 'sound/machines/hiss.ogg', 75, 1)
+	if(violent)
+		visible_message("[src] rapidly deflates!")
+		var/obj/item/inflatable/torn/R = new /obj/item/inflatable/torn(loc)
+		src.transfer_fingerprints_to(R)
+		qdel(src)
+	else
+		visible_message("[src] slowly deflates.")
+		spawn(50)
+			var/obj/item/inflatable/R = new /obj/item/inflatable(loc)
+			src.transfer_fingerprints_to(R)
+			qdel(src)
+
+/obj/structure/inflatable/verb/hand_deflate()
+	set name = "Deflate"
+	set category = "Object"
+	set src in oview(1)
+
+	if(usr.stat || usr.restrained())
+		return
+
+	deflate()
+
+/obj/item/inflatable/door
+	name = "inflatable door"
+	desc = "A folded membrane which rapidly expands into a simple door on activation."
+	icon = 'icons/obj/inflatable.dmi'
+	icon_state = "folded_door"
+
+/obj/item/inflatable/door/attack_self(mob/user)
+	playsound(loc, 'sound/items/zip.ogg', 75, 1)
+	to_chat(user, "<span class='notice'>You inflate [src].</span>")
+	var/obj/structure/inflatable/door/R = new /obj/structure/inflatable/door(user.loc)
+	src.transfer_fingerprints_to(R)
+	R.add_fingerprint(user)
+	qdel(src)
+
+/obj/structure/inflatable/door //Based on mineral door code
+	name = "inflatable door"
+	density = TRUE
+	anchored = TRUE
+	opacity = FALSE
+
+	icon = 'icons/obj/inflatable.dmi'
+	icon_state = "door_closed"
+
+	var/state = 0 //closed, 1 == open
+	var/isSwitchingStates = FALSE
+
+/obj/structure/inflatable/door/attack_ai(mob/user as mob) //those aren't machinery, they're just big fucking slabs of a mineral
```

Comment:
copypaste falling out of your pockets, bby >:C

Reply:
I just ported it